enhancements/network/on-prem-configurable-lb-dns-stack.md

```diff
@@ -0,0 +1,254 @@
+---
+title: on-prem-configurable-lb-dns-stack
+authors:
+  - "@yboaron"
+reviewers:
+  - "@sttts"
+  - "@deads2k"
+  - "@celebdor"
+  - "@bcrochet"
+  - "@bnemec"
+approvers:
+  - "@"
+creation-date: 2020-02-11
+last-updated: 2020-02-11
+status: 
+see-also:
+replaces:
+superseded-by:
+
+---
+
+# On-prem: configurable DNS/LB self-hosted stack 
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+In the current implementation,  the self-hosted DNS/LB stack runs under the MCO umbrella by default in all on-prem platforms and there's no option to configure it. 
+Some customers with high production needs usually have their own external Load Balancing and DNS resolution. In addition, there are cases in which some traffic used to provide these self-hosted services is disallowed in the customer's network.                                                                                                                                                                                                                                        The outcome of such scenarios is that cluster's resources are spent on providing unused services and generating traffic against certain customers network design.
```

Comment:
am curious, I can understand why load balancing might be out-of-cluster. But why DNS? Is this about the DNS resolution of Kube service URLs?

Reply:
Short answer: to avoid multicast traffic (generated by mDNS)

The full context:

The self-hosted infra DNS provides the following capabilities:
- node name resolution 
- resolve .apps , api , and api-int  names

The node name resolution is implemented by running mDNS-publisher (each node advertises himself)  and CoreDNS-mDNS plugin - means that every period of time a IP multicast packet is sent by each node.

There's a request from some customers to avoid using multicast traffic, they suggested providing node names resolution by their DNS (so we can drop mDNS) and even resolution of  .apps , api , and api-int  names . so in the latter case, we can even remove coreDNS pods and free the resources.